app/models/user.rb

```diff
@@ -83,6 +83,9 @@ def self.to_csv
   scope :unconfirmed, -> { where(confirmed_at: nil) }
   scope :locked_out, -> { where.not(locked_at: nil) }
 
+  # @return [Integer] Number of users who have added at least one note, given the note body is not empty
+  scope :has_notes, -> { joins(:notes).distinct.count { |user| user.notes.any? { |note| !note.body_is_empty } } }
```

Comment:
We can take it further. What if we wanted to know if any given user had an active learning log for example. Could we get to the point where we no longer need a block within a block? Have you seen this syntax before `things(&:foo?)`

Reply:
OK, more like this?

Reply:
OK, more like this? [94ddd97](https://github.com/DFE-Digital/early-years-foundation-recovery/pull/622/commits/94ddd972d48d2f4bf6dea8dbcb3b992fa2fbff40)